homework/src/index.js

```diff
@@ -30,17 +32,109 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+  async function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('p', header, {html: 'HYF Repositories', class: 'header-lable'});
+    const select = createAndAppend('select', header, { id: 'selectMenu' });
+    // use promise object
+    try {
+      const data = await fetchJSON(url);
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+        data.forEach((repository, index) => {
+          createAndAppend('option', select, {html: repository.name, value: index});
+        });
+        const container = createAndAppend('div', root, { id: 'container'});
+        contentLeft(container, data, select);
+        contentRight(container, data, select);
+    }
+    catch(error) {
+      createAndAppend('div', root, { html: error.message, class: 'alert-error' });
+    }
+  }
+  
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
+    const leftSection = createAndAppend('div', container, { id: 'leftSection', class: 'whiteframe' });
+    showTableRepo(leftSection, data[0]);
+
+    select.addEventListener('change', (event) => {
+      leftSection.innerHTML = "";
+      const theEventIndex = event.target.value;
+      showTableRepo(leftSection, data[theEventIndex]);
+    });
+  }
+
+  // drow the table
+function showTableRepo(leftSection, repository) {
+  // create table of the contents
+  const lTable = createAndAppend('table', leftSection);
+  // line one
+  const tr1 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr1, {html: 'Repository :', class: 'lable'});
+  const td1 = createAndAppend('td', tr1);
+  const nameLink = '<a href="' + repository.html_url + '" target="_blank">' + repository.name + '</a>';
+  createAndAppend('p', td1, {html: nameLink });
```

Comment:
nothing wrong with this, But if we didn't ues the `innerHTML` you couldn't hard coded this like so.
the only thing you can do is to switch:
```javascript
const nameLink = createAndAppend('p', td1);
createAndAppend('a', nameLink, {
  html: repository.name,
  href: repository.html_url,
  target: "_blank"
});
```